### PR TITLE
Implements Ling HUD

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Implements TG's Ling HUD.

This was never implemented when @TheDZD ported over TG style lings; being put in now.

What is displayed?
- Your current chemicals
- Your current active sting
 - Clicking on the sting icon will retract it

pics:

img1

Side note: our HUD code is a mess and utter crap.

Vampire HUD
- Usable Blood now uses the same visual counter as the ling chemical display
- Total blood and usable blood now show up under the status tab

:cl: Fox McCloud
rscadd: Adds in a Changeling chemical counter display
rscadd: Adds in a Changeling sting counter display
rscadd: Vampire total blood and usable blood will now appear under status
tweak: Changes vampire blood display to be similar to ling chemical counter
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/life.dm [hunk 1]
```diff
@@ -1059,8 +1059,13 @@ var/global/list/brutefireloss_overlays = list("1" = image("icon" = 'icons/mob/sc
 	return
 
 /mob/living/carbon/human/handle_changeling()
-	if(mind && mind.changeling)
-		mind.changeling.regenerate()
+	if(mind && hud_used)
```
Comment (tigercat2000) [inline code/modules/mob/living/carbon/human/life.dm]: this is a bad place to put that check, changelings should continue working regardless of if they have a HUD or not

**Comment by Gummin** [comment]
yes

**Comment by Dumbdumn5** [comment]
:+1:

**Comment by Fox-McCloud** [comment]
Turns out, I was wrong about Vamp blood showing up under status (thanks for mentioning that @Tastyfish).

This PR does some more things:

- Total+usable blood under status
- Better graphic for usable blood counter

**Comment by Spacemanspark** [comment]
Can you show us the graphic for the blood counter?

**Comment by Fox-McCloud** [comment]
@Spacemanspark 

It looks exactly like the ling chemical display that  you see above, instead of like this: 

img1

**Pull request merged**